driver-sync/src/test/functional/com/mongodb/client/ChangeStreamProseTest.java

```diff
@@ -0,0 +1,203 @@
+import com.mongodb.client.model.changestream.ChangeStreamDocument;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonInt32;
+import org.bson.BsonString;
+import org.bson.Document;
+import org.junit.Before;
+import org.junit.Test;
+
+import static com.mongodb.ClusterFixture.isDiscoverableReplicaSet;
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static com.mongodb.ClusterFixture.serverVersionLessThan;
+import static java.util.Arrays.asList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+
+// See https://github.com/mongodb/specifications/tree/master/source/transactions-convenient-api/tests/README.rst#prose-tests
+public class ChangeStreamProseTest extends DatabaseTestCase {
+    private static final long START_TIME_MS = 1L;
+    private static final long ERROR_GENERATING_INTERVAL = 121000L;
+
+    @Before
+    @Override
+    public void setUp() {
+        assumeTrue(canRunTests());
+        super.setUp();
+
+        // create the collection before starting transactions
+        collection.insertOne(Document.parse("{ _id : 0 }"));
+    }
+
+    //
+    // Test that the ChangeStream continuously tracks the last seen resumeToken.
+    //
+    @Test
+    public void testChangeStreamTracksResumeToken() {
+        MongoCursor<ChangeStreamDocument<Document>> cursor = collection.watch().iterator();
+
+        try {
+            collection.insertOne(Document.parse("{x: 1}"));
+
+            BsonDocument initialResumeToken = cursor.next().getResumeToken();
+            assertNotNull(initialResumeToken);
+        } finally {
+            cursor.close();
+        }
+    }
+
+    //
+    // Test that the ChangeStream will throw an exception if the server response is missing the resume token (if wire version is < 8).
+    //
+    @Test
+    public void testMissingResumeTokenThrowsException() {
+        assumeTrue(serverVersionLessThan(asList(4, 0, 7)));
+        boolean exceptionFound = false;
+
+        MongoCursor<ChangeStreamDocument<Document>> cursor = collection.watch(asList(Aggregates.project(Document.parse("{ _id : 0 }"))))
+                .iterator();
+        try {
+            collection.insertOne(Document.parse("{ x: 1 }"));
+            assertNull(cursor.next());
+        } catch (MongoChangeStreamException e) {
+            exceptionFound = true;
+        } finally {
+            cursor.close();
+        }
+        assumeTrue(exceptionFound);
+    }
+
+    //
+    // Test that the ChangeStream will automatically resume one time on a resumable error (including not master)
+    // with the initial pipeline and options, except for the addition/update of a resumeToken.
+    //
+    @Test
+    public void testResumeOneTimeOnError() {
+        assumeTrue(serverVersionAtLeast(4, 0));
+        setFailPoint("getMore", 10107);
```

Comment:
To ensure the failpoint has been triggered, add an assertion before the insertion that ensures tryNext returns null.

Reply:
I have added the assertion.